src/Our.Umbraco.Vorto/Web/UI/App_Plugins/Vorto/js/vorto.js

```diff
@@ -192,6 +202,10 @@
             }
         });
 
+        $scope.$watch("model.value.values", function () {
```

Comment:
I'd probably call it at the end of the $scope.$watch("activeLanguage"... function I think. Give it a try and see how it works. If it doesn't seem to work right, go with what you have.